WebApplication/Backend/Repositorys/FeedbackRepository.cs

```diff
@@ -1,24 +1,154 @@
-﻿using Backend.Repository;
+﻿using health_clinic_class_diagram.Backend.Dto;
 using Model.Blog;
+using MySql.Data.MySqlClient;
 using System;
 using System.Collections.Generic;
 
 namespace WebApplication.Backend.Repositorys
 {
+    /// <summary>
+    /// This class does connection with MySQL database feedback table
+    /// </summary>
     public class FeedbackRepository
     {
+        private MySqlConnection connection;
+        public FeedbackRepository()
+        {
+            try
+            {
+                connection = new MySqlConnection("server=localhost;port=3306;database=mydb;user=Tanjaa;password=TanjaaD");
+                connection.Open();
+            }
+            catch (Exception e)
+            {
+            }
+        }
+        ///Tanja Drcelic RA124/2017
+        /// <summary>
+        ///Get all feedbacks from feedbacks table
+        ///</summary>
+        ///<returns>
+        ///true if sucessful,else false
+        ///</returns>
+        ///<exception>
+        ///if any exception occurs method will return null
+        ///</exception>
         internal List<Feedback> GetAllFeedbacks()
         {
-            throw new NotImplementedException();
+                string sql1 = "Select * from feedbacks";
+                MySqlCommand cmd1 = new MySqlCommand(sql1, connection);
+                MySqlDataReader rdr = cmd1.ExecuteReader();
+                List<Feedback> resultList = new List<Feedback>();
+                while (rdr.Read())
+                {
+                    Feedback entity = new Feedback();
+                    entity.SerialNumber = (string)rdr[4];
+                    entity.PatientId = (String)rdr[3];
+                    entity.Text = (String)rdr[0];
+                    entity.Date = Convert.ToDateTime(rdr[2]);
+                    entity.Approved = (Boolean)rdr[1];
+                    resultList.Add(entity);
+
+                }
+                connection.Close();
+                return resultList;
+        }
+
+        internal List<Feedback> GetFeedbacks(Boolean approvedF)
```

Comment:
Is approvedF a typo or is it short for something?

Reply:
You could also refactor the entire class:
Extract the code that connects to the data base to one function that accepts sql select command as a parameter. That way you could just call that method from each of the getters (getAll, getApproved, getDisapproved) with an appropriate sql query 😄